### PR TITLE
Bump dependencies, take 2

### PR DESCRIPTION
**Pull request by nickgros**
Re-include the changes in #354 plus

- move react-switch to "dependencies"
- add "prop-types" as a dependency
- upgrade SVGR plugins (must now use @svgr/plugin-jsx)
- simplify vite.config.ts files

### CODE REVIEW

File: packages/synapse-react-client/package.json [hunk 1]
```diff
@@ -78,6 +76,7 @@
     "mui-one-time-password-input": "1.1.0",
     "plotly.js-basic-dist": "^2.24.3",
     "pluralize": "^8.0.0",
+    "prop-types": "^15.8.1",
```
Comment (nickgros) [inline packages/synapse-react-client/package.json:79]: For some reason, `tsup` isn't bundling/properly writing a dynamic import for the `prop-types` dependency, which is imported via `react-switch` (but also many other dependencies?)

This issue resolves if we include it in our dependencies.

File: packages/synapse-react-client/package.json [hunk 2]
```diff
@@ -95,7 +94,7 @@
     "react-router-dom": "^5.3.4",
     "react-select": "^5.7.3",
     "react-sizeme": "^3.0.2",
-    "react-test-renderer": "^18.2.0",
+    "react-switch": "^7.0.0",
```
Comment (nickgros) [inline packages/synapse-react-client/package.json:97]: move from devDependencies

File: packages/synapse-react-client/package.json [hunk 3]
```diff
@@ -95,7 +94,7 @@
     "react-router-dom": "^5.3.4",
     "react-select": "^5.7.3",
     "react-sizeme": "^3.0.2",
-    "react-test-renderer": "^18.2.0",
```
Comment (nickgros) [inline packages/synapse-react-client/package.json:101]: remove from dependencies (it's already in devDependencies)

File: packages/synapse-react-client/esbuild.config.mjs [hunk 4]
```diff
@@ -45,6 +45,7 @@ const esBuildOptions = {
       loadPaths: ['../../node_modules', 'node_modules'],
     }),
     svgrPlugin({
+      plugins: ['@svgr/plugin-jsx'],
```
Comment (nickgros) [inline packages/synapse-react-client/esbuild.config.mjs:48]: Upgraded svgr plugins use new versions of svgr which require specifying the plugins we use. We just use the `jsx` plugin.

**Review by jay-hodgson — APPROVED** [review]
Additional changes also look ok.  Thank you for tracking this down!

**Pull request merged**